### PR TITLE
refactor: fetch from store instead of api

### PR DESCRIPTION
**Pull request by dated**
## Proposed changes

Refactors in multiple places to lookup delegate information from the store instead of fetching from the API.

## Types of changes

- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ghost** [comment]
**Thanks for submitting this pull request!** A maintainer will review this in the next few days and explicitly select labels so you know what's going on.

_If no reviewer appears after a week, a reminder will be sent out._

**Comment by ghost** [comment]
A maintainer will review this in the next few days and explicitly select labels so you know what's going on.

_If no reviewer appears after a week, a reminder will be sent out._

**Comment by ghost** [comment]
The [ci/circleci: test-node-10](https://circleci.com/gh/dated/ark-explorer/1016?utm_campaign=vcs-integration-link&amp;utm_medium=referral&amp;utm_source=github-build-link) job is failing as of [9011f231824353e4bdb32060fb78af0634ca49fe](https://github.com/ArkEcosystem/explorer/commit/9011f231824353e4bdb32060fb78af0634ca49fe). Please [review the logs for more information](https://circleci.com/gh/dated/ark-explorer/1016?utm_campaign=vcs-integration-link&amp;utm_medium=referral&amp;utm_source=github-build-link).

_Once you've pushed the fixes, the build will automatically re-run. Thanks!_

File: src/store/modules/delegates.js [hunk 1]
```diff
@@ -28,8 +31,21 @@ export default {
...
+
   getters: {
     delegates: state => state.delegates,
-    forged: state => state.forged
+    forged: state => state.forged,
+
+    byPublicKey: (state) => publicKey => {
+      return state.delegates.find(delegate => {
+        return delegate.publicKey === publicKey
+      }) || false
```
Comment (j-a-m-l) [inline src/store/modules/delegates.js]: I'd use `null` (or even `undefined`) instead of `false` to not give the impression that the result is a boolean.

**Review by j-a-m-l — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Your pull request needs some changes. Please wait for a comment from one of our developers for more information.

File: src/components/wallet/Delegate.vue [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div v-show="Object.keys(delegate).length">
+  <div v-show="delegate">
```
Comment (j-a-m-l) [inline src/components/wallet/Delegate.vue]:
```suggestion
  <div v-if="delegate">
```

to avoid `TypeError`s (`v-show` renders always, but it is hidden).

**Review by j-a-m-l — CHANGES_REQUESTED** [review]
I've added a suggestion to fix some errors.

**Review by j-a-m-l — APPROVED** [review]

**Comment by ghost** [comment]
A member has approved this PR. A maintainer will merge this PR shortly. If it shouldn't be merged yet, please leave a comment saying so and we'll wait.

_Thank you for your contribution!_

**Comment by ghost** [comment]
Your pull request has been merged but was not assigned a bounty tier. A maintainer will assign a bounty tier to this pull request in the next few days.

**Comment by ghost** [comment]
Your pull request has been merged and marked as tier 3. _It will earn you $25 USD._

**Pull request merged**